### PR TITLE
[docs] Add more examples to sui client

### PR DESCRIPTION
**Pull request by stefan-mysten**
## Description 

Adds a few more examples to sui client.

## Test plan 

How did you test the new or updated feature?

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

File: docs/content/references/cli/client.mdx [hunk 1]
```diff
@@ -188,6 +188,211 @@ $ sui client dynamic-field 0x5
 ╰─────────────┴───────────────────────────────────────────────────────────────────────────────────────────╯
 ```
 
+### Send SUI or objects
+
+In this example, let's see how to transfer SUI or transfer an object from one address to another. First all, there two main commands for sending SUI or transfering objects: Pay and Transfer.
```
Comment (amnn) [inline docs/content/references/cli/client.mdx:193]:
```suggestion
In this example, let's see how to transfer SUI or transfer an object from one address to another. First of all, there are two main commands for sending SUI or transferring objects: Pay and Transfer. 
```

If "Pay" and "Transfer" are referring to commands, let's use `pay` and `transfer`.

File: docs/content/references/cli/client.mdx [hunk 2]
```diff
@@ -188,6 +188,211 @@ $ sui client dynamic-field 0x5
 ╰─────────────┴───────────────────────────────────────────────────────────────────────────────────────────╯
 ```
 
+### Send SUI or objects
+
+In this example, let's see how to transfer SUI or transfer an object from one address to another. First all, there two main commands for sending SUI or transfering objects: Pay and Transfer. 
+Each of these commands, has some sister commands: PaySui, PayAllSui, and TransferSui. The main difference between Pay and Transfer is that Pay allows you to send coin to multiple recipients, whereas Transfer only accepts one recipient. Another important difference is that Pay and Transfer require a gas object that is different than the one used to send the coins, which is used to pay gas fees.
```
Comment (amnn) [inline docs/content/references/cli/client.mdx:194]: Ditto on formatting of command names -- let's use the form that's recognisable from the CLI (e.g. `pay-sui`, `pay-all-sui` etc). Also, I would describe the differences between these commands as follows:

- Commands that end in `-sui` deal with Sui's native coin, and they use the input coins to pay for gas and for transferring.
- `pay-` commands typically deal with coins and handle smashing for you, whereas `transfer-` commands can handle the transfer of any object that has public transfer (has the `store` ability).

Some special cases:
- `pay-all-sui` is a special case of `pay-sui` that offers a way to transfer the entire balance after smashing.
- `transfer-sui` is a legacy command -- it has been entirely superseded by `pay-sui` or `pay-all-sui` depending on whether an amount is specified or not.

**Review by amnn — APPROVED** [review]
Thanks @stefan-mysten -- left some clarifying comments about the distinction between these commands, but this looks good.

File: docs/content/references/cli/client.mdx [hunk 3]
```diff
@@ -188,6 +188,211 @@ $ sui client dynamic-field 0x5
...
+│ 0x56d76420a5bc7d356e3930e6a2ddc61cbbc0c87ea5c5f3cc3ac5952b4d52be5a │ 200000000000       │ 200.00           │
+│ 0xc9b447fff18f13fa035e028534b8344d5fc8a8760248fad10155e78f44dc3a52 │ 200000000000       │ 200.00           │
+│ 0xe373de9bdbd3dbd4f5f707f144c83af74a181cdb3903a953ee366f48e33865c7 │ 200000000000       │ 200.00           │
+│ 0xf7adb6aeb18eb409c03fe88cc31711b61a65333e0fdd88c1815d4ec75f713f31 │ 200000000000       │ 200.00           │
+╰────────────────────────────────────────────────────────────────────┴────────────────────┴──────────────────╯
+```
+
+We want to send 0.5 SUI to `eloquent-amber`. Given that we have a few gas coins, we can use Pay. If we would have had only one gas coin, then we could have only used TransferSui, or PaySui, or we would have needed to split the coin first to have another coin to use for paying gas. In this case, let's use the TransferSui command as we will not need to provide a gas coin to be used for the gas fees. We need to pass the address (or its alias) of the address to which SUI is sent, which coin to use to transfer SUI from, and the amount of SUI to transfer.
```
Comment (ronny-mysten) [inline docs/content/references/cli/client.mdx:223]:
```suggestion
You want to send 0.5 SUI to `eloquent-amber`. Given that you have a few gas coins, you can use `pay`. If you had only one gas coin, then you would need to use `transfer-sui` or `pay-sui`, or you would need to split the coin to have another coin available for paying gas. In this case, let's use the `transfer-sui` command as you don't need to provide a gas coin for the gas fees. You need to pass the address (or its alias) to receive SUI, the coin to transfer SUI from, and the amount of SUI to transfer.
```

File: docs/content/references/cli/client.mdx [hunk 4]
```diff
@@ -188,6 +188,211 @@ $ sui client dynamic-field 0x5
...
+│  ┌──                                                                                              │
+│  │ Owner: Account Address ( 0x3d78abc33ccd173c521b4f9e78b21ea2c989960b631732c539efaa38bda30976 )  │
+│  │ CoinType: 0x2::sui::SUI                                                                        │
+│  │ Amount: 500000000                                                                              │
+│  └──                                                                                              │
+╰───────────────────────────────────────────────────────────────────────────────────────────────────╯
+
+```
+
+We can check the gas for the sender and the recipient's coins using `sui client gas <address>`. The sender now has 199.49 SUI for that gas coin that was used. 0.5 SUI was transferred, and the rest 0.01 SUI were the gas fees.
```
Comment (ronny-mysten) [inline docs/content/references/cli/client.mdx:356]:
```suggestion
You can check the gas for the sender and the recipient's coins using `sui client gas <address>`. The sender now has 199.49 SUI for that gas coin that was used. 0.5 SUI was transferred, and the remaining 0.01 SUI paid the gas fees.
```

File: docs/content/references/cli/client.mdx [hunk 5]
```diff
@@ -188,6 +188,211 @@ $ sui client dynamic-field 0x5
...
+
+sui client gas eloquent-amber
+╭────────────────────────────────────────────────────────────────────┬────────────────────┬──────────────────╮
+│ gasCoinId                                                          │ mistBalance (MIST) │ suiBalance (SUI) │
+├────────────────────────────────────────────────────────────────────┼────────────────────┼──────────────────┤
+│ 0xa031d3f8fd53b5f9885172e1c6bd8b770b25e42371c9ee215c6d4b21b2b73241 │ 500000000          │ 0.50             │
+╰────────────────────────────────────────────────────────────────────┴────────────────────┴──────────────────╯
+```
+
+If you'd like to send coins to multiple addresses, use one of the `Pay*` commands.
```
Comment (ronny-mysten) [inline docs/content/references/cli/client.mdx:377]:
```suggestion
If you'd like to send coins to multiple addresses, use one of the `pay*` commands.
```

File: docs/content/references/cli/client.mdx [hunk 6]
```diff
@@ -188,6 +188,211 @@ $ sui client dynamic-field 0x5
...
+
+If you'd like to transfer the whole object, you can use `sui client transfer-sui` and do not pass an amount:
```
Comment (ronny-mysten) [inline docs/content/references/cli/client.mdx:379]:
```suggestion
If you want to transfer the whole object, use `sui client transfer-sui` and do not pass an amount:
```

File: docs/content/references/cli/client.mdx [hunk 7]
```diff
@@ -188,6 +188,211 @@ $ sui client dynamic-field 0x5
...
+```
+sui client transfer-sui --to eloquent-amber --sui-coin-object-id 0xc9b447fff18f13fa035e028534b8344d5fc8a8760248fad10155e78f44dc3a52 --gas-budget 5000000
+```
+
+Then check again the gas for `eloquent-amber`
```
Comment (ronny-mysten) [inline docs/content/references/cli/client.mdx:384]:
```suggestion
Then check the gas for `eloquent-amber` again:
```

**Review by ronny-mysten — APPROVED** [review]
Noted some style things, but lgtm.

File: docs/content/references/cli/client.mdx (hunk 2 above)
Comment (stefan-mysten) [reply docs/content/references/cli/client.mdx:194]: Thanks, updated!

**Comment by stefan-mysten** [comment]
Thanks @ronny-mysten and @amnn! I implemented all the suggestions and hopefully this will clarify the differences and show examples on how to use these commands.

**Pull request merged**